### DISCUSSION TITLE
[3D] How to cast/receive Shadow using GltfSetMaterial OP?

### DISCUSSION THREAD

**Discussion by arrddon** [cables-gl/cables_docs#444] [Answers & Questions] [answered]
Hello, I have problem with shadow in 3D system when using 'GltfSetMaterial' OP.
Please let me know any solution to solve this problem.

![그림1](https://user-images.githubusercontent.com/24363682/210221290-35d2f968-c324-484b-b0cf-aff5a85fcc38.png)

**Comment by pandrr** [comment, marked as answer]
thanks for reporting!

i fixed a bugl it is now possible to do this on dev.cables.gl.
you need to use getshader to get the full modified shader

like this:

<img width="1035" alt="Screenshot 2023-01-04 at 10 14 48" src="https://user-images.githubusercontent.com/98792/210533794-521de291-a680-498f-aa37-75f07becc7e3.png">

**Reply by arrddon** [reply]
thank you! now it works
